Automattic/wp-calypso
# Devdocs: Fix routing and sidebar

This pull request fixes https://github.com/Automattic/wp-calypso/issues/21158 to prevent the url from being reverted when navigating to the `Login` page from the `Devdocs` page. It also fixes the sidebar that wouldn't appear when logged-out.

<img width="736" alt="screenshot" src="https://user-images.githubusercontent.com/594356/35161342-342d6bb6-fd40-11e7-948f-f6d8e2c30fb1.png">

#### Testing instructions

1. Run `git checkout fix/devdocs` and start your server, or open a [live branch](https://calypso.live/?branch=fix/devdocs)
2. Open the [`Devdocs` page](http://calypso.localhost:3000/devdocs) in an incognito tab
3. Assert that the sidebar is displayed
4. Click the `Log In` link in the upper right corner
5. Assert that url points to the [`Login` page](http://calypso.localhost:3000/log-in?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fdevdocs)

#### Additional notes

This pull request also fixes wrong file permissions (/cc @tyxla and @yoavf).

#### Reviews
